Lib/Src/CUartCom.cpp

```diff
@@ -85,23 +85,39 @@ bool CUartCom::init(UART_HandleTypeDef *p_huart,
  */
 void CUartCom::startRx()
 {
-    HAL_UART_Receive_IT(mp_huart, &m_rx_char, 1);
+    HAL_UART_Receive_IT(mp_huart, &m_rx_byte, 1);
 }
 
 /**
- * @brief Add data to the TX Queue and start transmission if possible
+ * @brief Disable UART RX Interrupts
+ * @note used when using half-duplex
+ */
+void CUartCom::stopRx()
+{
+    HAL_UART_AbortReceive_IT(mp_huart);
+}
+
+/**
+ * @brief Add string to the TX Queue and start transmission if UART is IDLE
  * @param msg Message to send.
- * @return True if message was added to the queue and transmission started
- * successfully.
+ * @return True if message was added to the queue. If transmission is attempted,
+ * it returns true if it was successful.
```

Comment:
Too much meaning to the return variable. Why wouldn't transmission not be successful?